testing/integration/factory-reset.nix

```diff
@@ -0,0 +1,93 @@
+let
+  pkgs = import ../../pkgs { };
+  # images are created in the testScript
+  bootVolumeImage = "/tmp/factory-reset-boot-volume.raw";
+  persistentVolumeImage = "/tmp/factory-reset-persistent-volume.raw";
+in
+with pkgs.lib;
+pkgs.testers.runNixOSTest {
+  name = "Controller system calls";
+
+  nodes = {
+    playos = { config, ... }: {
+      imports = [
+        ../../base/volatile-root.nix
+      ];
+
+      config = {
+        virtualisation.qemu.options = [
+            "-hda" bootVolumeImage       # /dev/sda
+            "-hdb" persistentVolumeImage # /dev/sdb
+        ];
+
+        # Needed to avoid virtualisation overriding volatile-root.nix fileSystem
+        # configuration
+        virtualisation.fileSystems = mkForce { };
+        virtualisation.writableStore = false;
+        virtualisation.diskImage = null;
+        fileSystems = {
+            # this is normally set up automatically by qemu-vm.nix but
+            # need to do it manually due to `virtualisation.fileSystems = {}`
+            "/nix/store" = {
+              device = "nix-store";
+              fsType = "9p";
+              neededForBoot = true;
+              options = [
+                "trans=virtio" "version=9p2000.L"
+                "msize=${toString config.virtualisation.msize}"
+                "cache=loose"
+              ];
+            };
```

Comment:
Are you referring to the `fileSystems."/nix/store"` entry specifically? Then the answer is "no", because [qemu-vm.nix overrides all `fileSystems` with `virtualisation.fileSystems`](https://github.com/NixOS/nixpkgs/blob/96fd503b3a7856ed964e4c81e9ba89dbc81367ee/nixos/modules/virtualisation/qemu-vm.nix#L1163) unless it is unset (like here), but that implies you have to set up everything manually.